converter.go

```diff
@@ -2162,6 +2168,38 @@ func arrowToRecordSingleColumn(ctx context.Context, field arrow.Field, col arrow
 	return newCol, nil
 }
 
+// returns a pointer to a new array which will be populated if we converted the array to valid utf8
+// or returns null, and an indicator that we should retain the original column
+func arrowStringRecordToColumn(
+	ctx context.Context,
+	stringCol *array.String,
+	mem memory.Allocator,
+	numRows int64,
+	fieldMetadata fieldMetadata,
+) (*arrow.Array, retainOriginalCol) {
```

Comment:
If I remember correctly - we don't need to return pointer here. Arrow already handles memory efficiently, and returning `arrow.Array` should not do memory copying.

Reply:
Confirmed that you are correct, updated to not return a pointer